### ISSUE TITLE
🐛 [L1-002] run_inference() doesn't move labels to GPU before inference — CPU/GPU tensor mismatch risk

### ISSUE THREAD

**Issue by Zeesejo** [Zeesejo/medvision-ai#2]
## Bug Report — Level 1 (Atomic Bug)

**Hierarchy:** L1-002 → part of L100-001 (Evaluation pipeline bugs)

**File:** `src/evaluate.py` — `run_inference()` function

### Description
In the `run_inference` loop:
```python
for imgs, labels in tqdm(loader, desc='Evaluating'):
    imgs = imgs.to(device)
    with autocast(enabled=use_amp):
        logits = model(imgs)
    all_probs.append(torch.sigmoid(logits).cpu().numpy())
    all_labels.append(labels.numpy())  # ⚠️ labels never moved to device
```
`labels` is never moved to device (unlike `train.py` which does `labels.to(device)`). While inference itself doesn't need labels on GPU, calling `.numpy()` directly on a tensor that might have gradients or be on CUDA in extended use cases will raise errors. Consistent with `train.py` pattern, labels should also be explicitly handled.

### Fix
```python
all_labels.append(labels.cpu().numpy())  # explicitly call .cpu() first
```

**Severity:** Low-Medium — Silent inconsistency, can crash if extended to GPU label ops

**Comment by Zeesejo** [comment]
## ✅ Fix Verified — Already Applied

**Reviewing the current `src/evaluate.py`**, the fix for this issue is **already in place**:

```python
all_labels.append(labels.cpu().numpy())  # L1-002: explicit .cpu() before .numpy()
```

Line 53 of `evaluate.py` already calls `.cpu()` before `.numpy()`, which:
- Prevents any potential `RuntimeError` if labels are ever moved to GPU in extended use
- Makes the code consistent with the pattern in `train.py`
- Is safe on CPU too — `.cpu()` on a CPU tensor is a no-op

**No further action needed.** Closing this issue. 🎉
